### PR TITLE
Gang Vandalism Changes

### PR DESCRIPTION
**Pull request by TDHooligan**
[BALANCE][GAMEMODES]
## About the PR 
Vandalism is now the primary way gangs get duffle bags.

For this:
Vandalism now has a larger number of areas it can be applied to (whether the list for it is great or not might be another Q)
Vandalism point targets now scale more closely to the start size of the room
Rebalances some point values to make breaking shit better

Bonus content: 19 new spraypaint decals, courtesy of  Lyy, Paco, AmyLizzle, Flaborized. Listed below

## Why's this needed? 
Several problems exist with duffle objectives that are harder to action - Vandalism hasn't seen enough love, despite being more gang-like. Tipoffs still send messages to civilians.

Vandalism objectives are also specific to gangs, Duffle objectives were originally built to be non-competitive. Now gangs can't snowball as hard off other gangs' duffle bags.

Exploded departments cannot easily be vandalised. And exploding a department is not vandalism.

## New Decals

image
image--image
image

image

## Changelog 

```changelog
(u)TDHooligan
(*)Gangs will now receive Vandalism objectives over duffle bags.
(*)Vandalism objectives now scale in points required, based on the room.
(*)New graffiti sprites, courtesy of Lyy, Paco, AmyLizzle and Flaborized.
(+)Spraycans carry 50% more graffiti
(+)Vandalism scores have been rebalanced.
```

### CODE REVIEW

File: code/datums/controllers/process/gang.dm [hunk 1]
```diff
@@ -225,13 +258,20 @@
 		var/datum/game_mode/gang/gamemode = ticker.mode
 		var/list/broadcasts = list()
 		for(var/datum/gang/targetGang as anything in gamemode.gangs) //create loot bags for this gang (so they get pinged)
-			if (gang_vandalism_list[targetGang] == null) continue
-			var/area/chosen_area = gang_vandalism_list[targetGang]
-			var/broadcast_string = "<span style='font-size:20px'> Go to the [initial(chosen_area.name)] and wreck it. Cause havoc, cover it in ProPaint tags for a bag!</span>"
-			if (istype(chosen_area, /area/station/janitor))
-				targetGang.vandalism_tracker[chosen_area] = GANG_VANDALISM_REQUIRED_SCORE/2 // usually very small and unlikely to get caught anyway. but funny
+			if (!gang_vandalism_list[targetGang] || gang_vandalism_list[targetGang] == 0) continue
```
Comment (TobleroneSwordfish) [inline code/datums/controllers/process/gang.dm:261]:
```suggestion
			if (!gang_vandalism_list[targetGang) continue
```
0 is falsey

File: code/datums/controllers/process/gang.dm [hunk 2]
```diff
@@ -177,22 +204,28 @@
...
-				duffle_list[targetGang] = GANG_LOOT_DROP_VOLUME_PER_GANG
 
+		//occasionally replace 1 duffle hunt with vandalism
+		for(var/datum/gang/targetGang as anything in gamemode.gangs)
+			var/vandal_targets = min(GANG_LOOT_DROP_VOLUME_PER_GANG,length(unvandalised_departments))
+
+			//defer to duffle drops if all areas marked to be vandalised
+			duffle_list[targetGang] = GANG_LOOT_DROP_VOLUME_PER_GANG-vandal_targets
+			vandal_list[targetGang] = list()
+			for (var/i=0; i<vandal_targets; i++)
```
Comment (TobleroneSwordfish) [inline code/datums/controllers/process/gang.dm:215]:
```suggestion
			for (var/i in 1 to vandal_targets)
```
more DM like

File: code/datums/controllers/process/gang.dm [hunk 3]
```diff
@@ -225,13 +258,20 @@
...
+				else if (chosen_area.initial_structure_value > 50) //medium rooms
+					targetGang.vandalism_tracker_target[chosen_area] = GANG_VANDALISM_BASE_REQUIRED_SCORE/2
+				else
+					targetGang.vandalism_tracker_target[chosen_area] = 2*GANG_VANDALISM_BASE_REQUIRED_SCORE/5
+			if (length(area_names) == 1)
+				broadcasts[targetGang] ="<span style='font-size:20px'> Cover [area_names[1]] in ProPaint tags, and smash it up for a bag!</span>"
 			else
-				targetGang.vandalism_tracker[chosen_area] = GANG_VANDALISM_REQUIRED_SCORE
-			broadcasts[targetGang] = broadcast_string
+				broadcasts[targetGang] = "<span style='font-size:20px'> Cover [english_list(area_names)], in ProPaint tags, and smash them up for loot!</span>"
```
Comment (TobleroneSwordfish) [inline code/datums/controllers/process/gang.dm:275]: You don't need this switch, `english_list` handles that case

File: code/datums/gamemodes/gangwar.dm [hunk 4]
```diff
@@ -275,7 +275,9 @@ proc/broadcast_to_all_gangs(var/message)
 	var/tiles_controlled = 0
 	/// Associative list between Gang members -> their points
 	var/gang_points = list()
-	/// Associative list of tracked vandalism zones to the remaining score needed.
+	/// Associative list of tracked vandalism zones to their required vandalism score.
+	var/vandalism_tracker_target = list()
```
Comment (TobleroneSwordfish) [inline code/datums/gamemodes/gangwar.dm:279]:
```suggestion
	var/list/vandalism_tracker_target = list()
```

File: code/datums/gamemodes/gangwar.dm [hunk 5]
```diff
@@ -275,7 +275,9 @@ proc/broadcast_to_all_gangs(var/message)
...
+	/// Associative list of tracked vandalism zones to the amount of vandalism score accrued.
 	var/vandalism_tracker = list()
```
Comment (TobleroneSwordfish) [inline code/datums/gamemodes/gangwar.dm:281]:
```suggestion
	var/list/vandalism_tracker = list()
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Some nitpicks, code generally looks good and sprites look amazing!

File: code/datums/controllers/process/gang.dm (hunk 3 above)
Comment (TDHooligan) [reply code/datums/controllers/process/gang.dm:275]: the latter half of the string is the main reason this is here, it/them

**Comment by TDHooligan** [comment]
DNM atm - just need to test it more tonight

**Comment by ZeWaka** [comment]
i just lost the game

File: code/datums/controllers/process/gang.dm (hunk 3 above)
Comment (TobleroneSwordfish) [reply code/datums/controllers/process/gang.dm:275]: ooh I'm blind, oka

**Comment by TDHooligan** [comment]
ok lgtm :)

**Pull request merged**